backend/oasst_backend/utils/message_tree_topic_modeling.py

```diff
@@ -0,0 +1,96 @@
+import argparse
+
+from bertopic import BERTopic
+from bertopic.representation import MaximalMarginalRelevance
+from bertopic.vectorizers import ClassTfidfTransformer
+from exported_tree_loading import load_data
+from sentence_transformers import SentenceTransformer
+from similarity_functions import compute_cos_sim_kernel, embed_data, k_hop_message_passing_sparse
+from sklearn.feature_extraction.text import CountVectorizer
+
+parser = argparse.ArgumentParser(description="Process some arguments.")
+parser.add_argument("--model_name", type=str, default="all-MiniLM-L6-v2")
+parser.add_argument("--cores", type=int, default=1)
+parser.add_argument("--pair_qa", type=bool, default=True)
+parser.add_argument("--use_gpu", type=bool, default=False)
+parser.add_argument("--batch_size", type=int, default=128)
+parser.add_argument("--k", type=int, default=2)
+parser.add_argument("--threshold", type=float, default=0.65)
+parser.add_argument("--exported_tree_path", nargs="+", help="<Required> Set flag", required=True)
+# Use like python message_tree_topic_modeling.py --exported_tree_path 2023-02-06_oasst_prod.jsonl 2023-02-07_oasst_prod.jsonl
```

Comment:
This comment seems to be helpful .. could you please move it into a doc string to `__main__`? Also maybe put the argument parsing in a function that is called by `__main__` instead of having it always executed in global scope?

Reply:
moved comment to doc string in __main__

Reply:
moved them to a function and call it in __main__